File: flutter/lib/market_data/market_refresh_snapshot.dart
import "../domain/market_cache_models.dart";
import "market_fetch_queue.dart";

class MarketRefreshSnapshot {
  const MarketRefreshSnapshot({
    required this.quotesBySymbol,
    required this.historiesBySymbol,
    required this.progress,
    required this.lastUpdatedAtMs,
    required this.refreshErrorMessage,
  });

  final Map<String, CachedMarketQuote> quotesBySymbol;
  final Map<String, CachedHistoricalPrices> historiesBySymbol;
  final MarketFetchQueueProgress progress;
  final int? lastUpdatedAtMs;
  final String? refreshErrorMessage;

  static const MarketRefreshSnapshot empty = MarketRefreshSnapshot(
    quotesBySymbol: <String, CachedMarketQuote>{},
    historiesBySymbol: <String, CachedHistoricalPrices>{},
    progress: MarketFetchQueueProgress(totalCount: 0, completedCount: 0, running: false, currentLabel: null),
    lastUpdatedAtMs: null,
    refreshErrorMessage: null,
  );
}

File: swift/Sources/MoneyHero/MarketData/MarketRefreshSnapshot.swift
public struct MarketRefreshSnapshot: Sendable {
    public let quotesBySymbol: [String: CachedMarketQuote]
    public let historiesBySymbol: [String: CachedHistoricalPrices]
    public let progress: MarketFetchQueueProgress
    public let lastUpdatedAtMs: Int?
    public let refreshErrorMessage: String?

    public init(
        quotesBySymbol: [String: CachedMarketQuote],
        historiesBySymbol: [String: CachedHistoricalPrices],
        progress: MarketFetchQueueProgress,
        lastUpdatedAtMs: Int?,
        refreshErrorMessage: String?
    ) {
        self.quotesBySymbol = quotesBySymbol
        self.historiesBySymbol = historiesBySymbol
        self.progress = progress
        self.lastUpdatedAtMs = lastUpdatedAtMs
        self.refreshErrorMessage = refreshErrorMessage
    }

    public static let empty = MarketRefreshSnapshot(
        quotesBySymbol: [:],
        historiesBySymbol: [:],
        progress: MarketFetchQueueProgress(totalCount: 0, completedCount: 0, running: false, currentLabel: nil),
        lastUpdatedAtMs: nil,
        refreshErrorMessage: nil
    )
}
